### DISCUSSION TITLE
pathology DSA training

### DISCUSSION THREAD

**Discussion by hw-ju** [Project-MONAI/MONAILabel#1046] [Q&A]
Hi everyone! @diazandr3s
Can we have some doc/tutorial on how to use **MONAILabelTraining** in DSA, including how to set up parameters on the left panel, and which dataset on DSA will be trained on? Thanks!
<img src="https://user-images.githubusercontent.com/90806807/194181787-558a1bf6-b1b6-4f71-87fa-44a76418fd69.png" width=50% height=50%>
<img src="https://user-images.githubusercontent.com/90806807/194181866-756519e8-63cd-429a-ba06-f15d9482a349.png" width=30% height=30%>

**Comment by diazandr3s** [comment]
Hi @hw-ju,

Thanks for opening this discussion.

As far as I know, Girder API is part of the DSA project: https://digitalslidearchive.github.io/HistomicsTK/examples/introducing_the_girder_api.html

It helps organize/manage data, user accounts, and annotations.  Is there a use case you'd like to discuss using Girder?

Regarding this:

> which dataset on DSA will be trained on?

Similar to the Radiology App, the model is trained or updated on the labels user has submitted after editing or creating them on the viewer.

Hope this helps,

**Reply by alex-virodov** [reply]
Hello @diazandr3s, I think this thread is a result of my questions to @hw-ju during our meeting.

> the model is trained or updated on the labels user has submitted after editing or creating them on the viewer.

I have assumed that's the case, but I did not see the code implementing this. Specifically, I assumed that I can add a bunch of rectangle/poly annotations in DSA with appropriate labels (my example below), and get the underlying models to segment tissue based on those annotations. However, I did not see any code reading annotations from DSA. 

So my question is - how would I go about preparing labels? If "creating them on the viewer", how does monai-label know which annotations to pull from DSA? And does it pull them? Or is there a different workflow?

Hopefully I am missing some obvious flow, sorry if so. If not, and this functionality is not yet implemented, I would be happy to work on it. I made a somewhat related PR for an interactive NuClick DSA integration (https://github.com/Project-MONAI/MONAILabel/pull/1058).

My "ideal" annotation workflow:
![image](https://user-images.githubusercontent.com/3259999/194947976-8e287b5c-7fd4-4052-b703-de5fdf824874.png)

== Background ==

First, thank you for your videos on Monai-Label, I learned much from them. 

My name is Alex and I am a software developer/data scientist at the University of Kentucky. We have some pathology tissue-level segmentation tasks (H&E, Ki67) and I would like to use DSA & Monai-label as a platform for that.

Thank you.
Alex.

**Reply by diazandr3s** [reply]
Hi @alex-virodov,

It is great to have your contribution. Nice to meet you as well :)

[MONAI Label plugin for DSA](https://github.com/Project-MONAI/MONAILabel/tree/main/plugins/dsa) makes a POST REST API call for model training/updating using the available annotations (https://github.com/Project-MONAI/MONAILabel/blob/main/plugins/dsa/cli/client.py#L202) - Of course, if there are annotations either created by MONAI Label or manually, 

Here is a video showing annotation and training using DSA.

https://user-images.githubusercontent.com/11991079/194974636-d44c2df4-b484-405f-bde3-f5451e37ef5f.mp4

> So my question is - how would I go about preparing labels? If "creating them on the viewer", how does monai-label know which annotations to pull from DSA? And does it pull them? Or is there a different workflow?

Every time you click on training functionality on DSA ( meaning a POST REST API call), annotations and images are sent to the MONAI Label server for training. @SachidanandAlle please keep me honest here.

Hope this makes sense. Please let us know if it is unclear.
